lib/restforce/middleware/custom_headers.rb

```diff
@@ -0,0 +1,12 @@
+module Restforce
+  # Middleware that allows you to specify custom request headers
+  # when initializing Restforce client
+  class Middleware::CustomHeaders < Restforce::Middleware
+    def call(env)
+      headers = @options[:request_headers]
+      env[:request_headers].merge!(headers) if headers.is_a?(Hash)
```

Comment:
Would it make sense to merge the specified `request_headers` on top of `headers`? That would allow you to override which feels like the right behaviour.